src/entity/User.ts

```diff
@@ -135,6 +136,45 @@ const checkUsernameAndEmail = async (
   return !user;
 };
 
+type UpdateUserEmailResult =
+  | { status: 'ok'; userId: string }
+  | { status: 'failed'; reason: UpdateUserFailErrorKeys; error?: Error };
+export const updateUserEmail = async (
+  con: Connection,
+  data: UpdateUserEmailData,
+  logger: FastifyLoggerInstance,
+): Promise<UpdateUserEmailResult> => {
+  if (!data.email || !data.id) {
+    return { status: 'failed', reason: 'MISSING_FIELDS' };
+  }
+
+  return con.transaction(async (entityManager) => {
+    try {
+      await entityManager
+        .getRepository(User)
+        .update({ id: data.id }, { email: data.email });
+      logger.info(`Updated email for user with ID: ${data.id}`);
+      return { status: 'ok', userId: data.id };
+    } catch (error) {
+      logger.error(
+        {
+          data,
+          userId: data.id,
+          error,
+        },
+        'failed to create user profile',
+      );
+
+      // Unique
+      if (error?.code === '23505') {
```

Comment:
This error will be thrown only when you try to update the email to an already existing email.
If the user doesn't exist nothing will be thrown